src/features/linkApplicationToCustomer/LinkApplicationToCustomerContainer.tsx

```diff
@@ -61,11 +61,28 @@ const LinkApplicationToCustomerContainer = ({
     variables: filteredCustomersVars,
   });
 
+  const exactlyFilteredCustomersVars: FILTERED_CUSTOMERS_VARS = {
+    first: 1,
+    [SearchBy.FIRST_NAME]: application.firstName,
+    [SearchBy.LAST_NAME]: application.lastName,
+    [SearchBy.EMAIL]: application.email,
+  };
```

Comment:
This will not tackle the situation where multiple customers are created with the same data, for some reason. If there were two applications from the same new customer and different handlers would handle those at the same time, two profiles would be created.

But, in practice that should never happen, at least not before we get verified signups.

Reply:
Yes, to prevent this kind of race condition the check would have to be done on the server side. And as the creation actually does creation in two different backends through federation I don't see a reasonable solution to this without overhauling the architecture to properly support federated mutations.